### PR TITLE
Don't allow merging PR's which are being conflict checked

### LINKED ISSUE #19352: Merged pull not marked as merged when conflict checking still in progress
### Description

Merging a pull request on the web UI doesn't mark it as merged when the conflict checking is still in progres.

1. Merge PR A.
2. Navigate to PR B and merge it, while the conflict checking after the first merge is still in progress.
3. See that the PR is merged (merge commit exists in the target branch), but the PR is not marked as merged.
4. The already-merged PR B still shows the "conflict checking still in progress" information

I encountered this on Codeberg, here: https://codeberg.org/Codeberg/Documentation/pulls/211 (the PR was marked as manually merged, but the actual merge happend using the web UI). I couldn't reproduce on try.gitea.io, but I suppose it's a timing issue with the merge conflict checker, I didn't manage to get into a state where I could see the "Conflict checking still in progress window" (but I saw this on Codeberg after B was merged).

CC @Gusted 

### Gitea Version

1.16.5

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

image

### Git Version

_No response_

### Operating System

Debian GNU/Linux

### How are you running Gitea?

Codeberg deployment (manually compiled custom fork), run without container on a Debian Linux machine.

### Database

MySQL

### PR DESCRIPTION
**Pull request by Gusted**
- When a PR is still being conflict checked, don't allow the PR to be merged(the merge button could already be visible before e.g. a new commit was pushed to the PR).
- Resolves #19352

### CODE REVIEW

**Review by wxiaoguang — APPROVED** [review]

File: services/pull/check.go [hunk 1]
```diff
@@ -36,6 +36,7 @@ var (
 	ErrUserNotAllowedToMerge = errors.New("user not allowed to merge")
 	ErrHasMerged             = errors.New("has already been merged")
 	ErrIsWorkInProgress      = errors.New("work in progress PRs cannot be merged")
+	ErrIsChecking            = errors.New("PRs which are still being conflict checked cannot be merged")
```
Comment (delvh) [inline services/pull/check.go]: Alternatively,
```suggestion
	ErrIsChecking            = errors.New("cannot merge while conflict checking is in progress")
```
I'm fine with both.

**Review by delvh — APPROVED** [review]

File: services/pull/check.go [hunk 2]
```diff
@@ -88,6 +89,10 @@ func CheckPullMergable(ctx context.Context, doer *user_model.User, perm *models.
 		return ErrNotMergableState
 	}
 
+	if pr.IsChecking() {
```
Comment (lunny) [inline services/pull/check.go:92]: We still also need to check if it's in the checking queue.

**Review by lunny — CHANGES_REQUESTED** [review]
~I think it's not enough, checking queue check is also a necessary.~

File: services/pull/check.go (hunk 2 above)
Comment (Gusted) [reply services/pull/check.go:92]: The status that's being checked in `IsChecking` is the status for those PR's that are currently in the queue.

**Review by lunny — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Comment by pajamrazek** [comment]
Hi, we use Gitea 1.16.6 which should have fixed this. However, the bug is still there. UI sometimes hangs even dozens of minutes with "Merge conflict checking is in progress. Try again in few moments." Please could you check again?

**Pull request merged**